source/server/configuration_impl.cc

```diff
@@ -171,7 +172,30 @@ WatchdogImpl::WatchdogImpl(const envoy::config::bootstrap::v3::Watchdog& watchdo
   multikill_timeout_ =
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, multikill_timeout, 0));
   multikill_threshold_ = PROTOBUF_PERCENT_TO_DOUBLE_OR_DEFAULT(watchdog, multikill_threshold, 0.0);
-  actions_ = watchdog.actions();
+  auto actions = watchdog.actions();
+
+// Add abort_action if it's available on the given platform and killing is
+// enabled since it's more helpful than the default kill action.
+#ifndef WIN32
```

Comment:
I would think this part should be platform agnostic. And it will be up to the implementation of this events to handle cross platform differences.

Reply:
Ideally, yes, but the Abort Action implementation is currently marked to be excluded from Windows since it doesn't support Windows yet.

See the prior PR: https://github.com/envoyproxy/envoy/pull/12860#discussion_r487338546

Reply:
Sorry for missing the review on the initial PR, but did we consider on platforms like Windows to enable the extension but  fall back to the basic watchdog full process kill/panic behavior (and log the thread id to be killed and maybe more diagnostic info)?

We could use Bazel platform selection instead of ifdefs to build a small platform specific "thread kill" library and that would be localized to the abort action extension (and allow us to iterate further if we are able to support the extension fully on Windows).

Meta question: Not sure if the philosophy of the project is to just disable a feature/extension rather than offer a degraded experience on a different platform

Reply:
@sunjayBhatia yes, if you look at [this intermediate commit](https://github.com/envoyproxy/envoy/pull/12860/files/0b3c91c80d4e71f29cc662852848a2b119ae46ef#diff-8b081a122bce2e3ce14eff23bcbb9e39R46) I was originally just having windows print out unimplemented, and the default panic behavior would kill the process.

Regarding the philosophy, I've seen some other extensions that are also just disabled for windows -- some of which might just arise from lack of resources, but ultimately across platforms there are some differences in capabilities (one example being signals).

Any thoughts @envoyproxy/api-shepherds ?

Reply:
> I've seen some other extensions that are also just disabled for windows

Yeah, these are related to external dependencies that require a decent amount of work to get compiling on Windows, we haven't gotten around to those yet so we disabled the relevant extensions

Reply:
OK sorry I didn't realize that this is the direction we are going to go. Here is what I would do:
1) Per my other comment I would move the code back to core, since it's not really an extension anymore.
2) I would have a platform function to kill a thread.
3) On Windows just abort/panic the process as it did before. On Linux do what you have now. This should allow the extension to be compiled on all platforms and do something sane.

Does that work?

Reply:
SGTM. Will close this PR while making those changes,